src/NzbDrone.Core/Datastore/Extensions/SqlMapperExtensions.cs

```diff
@@ -1,20 +1,26 @@
 using System;
 using System.Collections.Generic;
 using System.Data;
+using System.Threading;
+using System.Threading.Tasks;
 using Dapper;
 
 namespace NzbDrone.Core.Datastore
 {
     public static class SqlMapperExtensions
     {
-        public static IEnumerable<T> Query<T>(this IDatabase db, string sql, object param = null)
+        public static async Task<IEnumerable<T>> QueryAsync<T>(
+            this IDatabase db,
+            string sql,
+            object param = null,
+            CancellationToken cancellationToken = default)
         {
-            using (var conn = db.OpenConnection())
+            using (var conn = await db.OpenConnectionAsync(cancellationToken).ConfigureAwait(false))
```

Comment:
Sadly Dapper sucks at making use of the cancellation token for queries, so `ThrowIfCancellationRequested` should be used in these circumstances which is the perfect use-case for it IMO.

The using must await-ed too. If you aren't already, you should switch to Rider as it's showing suggestions for these cases.

```suggestion
            cancellationToken.ThrowIfCancellationRequested();

            await using var conn = await db.OpenConnectionAsync(cancellationToken).ConfigureAwait(false);
```

Reply:
Sure, I’ll add the cancellation check. I’m deeply used to VS, it’s just where I’m most comfortable. When I’ve tried Rider on Linux, it always felt a bit off, and my Git workflow isn’t as smooth there. I’ll at least consider giving it another look, though.

But in general, is this the right direction? As when I prepare to transition the services, I want to avoid constantly rebasing onto all the ongoing changes. For example, the mini profiler is also moving from `IDbConnection` to `DbConnection` isn’t a big deal since it’s a small change. However, it’s a different story with the services they’re much more involved. I would like it to be merged in smaller PR's instead of one large PR.

Reply:
Sadly Rider is more annoying at making this kind of suggestions. 

Personally I consider these changes too big of a task, and should be instead in increments having sync and async methods available. Like I implemented async support for HttpClient just to add HTTP2 support, sync just calls the async due to the limitation.

Taking EventAggregator for example, you have one big issue as you have both Handle and HandleAsync, but isn't really async. And the TaskFactory without the Wait() makes me thing it's actually an attempt to make it a background handle actually. 

I noticed you added `IHandleBackgroundAsync` to address this, but at the moment I can't think of any suggestions in order to handle sync and real async handlers, maybe we should address it in the caller like wolverine does, PublishAsync doesn't wait since it's processed in the background, while InvokeAsync waits.

FYI in NotificationService the behavior is changed when you made all of them `IHandleBackgroundAsync`, while it builds Rider is throwing errors:
![Screenshot 2025-12-27 at 17 35 46](https://github.com/user-attachments/assets/51f18d24-9140-4d32-824f-d5d4298fa0f6)